diff -rc pwc-10.0.12-rc1/pwc-ctrl.c pwc-10.0.12-rc1-cw/pwc-ctrl.c
*** pwc-10.0.12-rc1/pwc-ctrl.c	2006-04-13 22:59:53.000000000 +0200
--- pwc-10.0.12-rc1-cw/pwc-ctrl.c	2007-10-26 02:59:01.000000000 +0200
***************
*** 42,47 ****
--- 42,48 ----
  #endif
  #include <asm/errno.h>
  #include <linux/version.h>
+ #include <media/v4l2-dev.h>
   
  #include "pwc.h"
  #include "pwc-ioctl.h"
diff -rc pwc-10.0.12-rc1/pwc-dec23.c pwc-10.0.12-rc1-cw/pwc-dec23.c
*** pwc-10.0.12-rc1/pwc-dec23.c	2006-04-13 23:30:32.000000000 +0200
--- pwc-10.0.12-rc1-cw/pwc-dec23.c	2007-10-26 03:00:41.000000000 +0200
***************
*** 882,888 ****
  
  		}	
  #else
! 		memcpy(dst, 0, pwc->view.x * pwc->view.y);
  #endif
  
  	} else {
--- 882,888 ----
  
  		}	
  #else
! 		memset(dst, 0, pwc->view.x * pwc->view.y);
  #endif
  
  	} else {
diff -rc pwc-10.0.12-rc1/pwc.h pwc-10.0.12-rc1-cw/pwc.h
*** pwc-10.0.12-rc1/pwc.h	2006-04-13 23:38:43.000000000 +0200
--- pwc-10.0.12-rc1-cw/pwc.h	2007-10-26 02:51:06.000000000 +0200
***************
*** 25,31 ****
  #ifndef PWC_H
  #define PWC_H
  
! #include <linux/config.h>
  #include <linux/module.h>
  #include <linux/usb.h>
  #include <linux/spinlock.h>
--- 25,31 ----
  #ifndef PWC_H
  #define PWC_H
  
! #include <linux/autoconf.h>
  #include <linux/module.h>
  #include <linux/usb.h>
  #include <linux/spinlock.h>
diff -rc pwc-10.0.12-rc1/pwc-if.c pwc-10.0.12-rc1-cw/pwc-if.c
*** pwc-10.0.12-rc1/pwc-if.c	2006-04-11 22:22:09.000000000 +0200
--- pwc-10.0.12-rc1-cw/pwc-if.c	2007-10-26 03:05:03.000000000 +0200
***************
*** 65,70 ****
--- 65,71 ----
  #include <linux/version.h>
  #include <asm/io.h>
  #include <linux/moduleparam.h>
+ #include <media/v4l2-dev.h>
  
  #include "pwc.h"
  #include "pwc-ioctl.h"
***************
*** 707,713 ****
  /* This gets called for the Isochronous pipe (video). This is done in
   * interrupt time, so it has to be fast, not crash, and not stall. Neat.
   */
! static void pwc_isoc_handler(struct urb *urb, struct pt_regs *regs)
  {
  	struct pwc_device *pdev;
  	int i, fst, flen;
--- 708,714 ----
  /* This gets called for the Isochronous pipe (video). This is done in
   * interrupt time, so it has to be fast, not crash, and not stall. Neat.
   */
! static void pwc_isoc_handler(struct urb *urb)
  {
  	struct pwc_device *pdev;
  	int i, fst, flen;
diff -rc pwc-10.0.12-rc1/pwc-uncompress.h pwc-10.0.12-rc1-cw/pwc-uncompress.h
*** pwc-10.0.12-rc1/pwc-uncompress.h	2006-01-07 12:57:47.000000000 +0100
--- pwc-10.0.12-rc1-cw/pwc-uncompress.h	2007-10-26 02:51:19.000000000 +0200
***************
*** 30,36 ****
  #ifndef PWC_UNCOMPRESS_H
  #define PWC_UNCOMPRESS_H
  
! #include <linux/config.h>
  
  #include "pwc-ioctl.h"
  
--- 30,36 ----
  #ifndef PWC_UNCOMPRESS_H
  #define PWC_UNCOMPRESS_H
  
! #include <linux/autoconf.h>
  
  #include "pwc-ioctl.h"
  
diff -rc pwc-10.0.12-rc1/pwc-v4l.c pwc-10.0.12-rc1-cw/pwc-v4l.c
*** pwc-10.0.12-rc1/pwc-v4l.c	2006-03-01 17:59:06.000000000 +0100
--- pwc-10.0.12-rc1-cw/pwc-v4l.c	2007-10-26 03:02:03.000000000 +0200
***************
*** 33,38 ****
--- 33,39 ----
  #include <linux/slab.h>
  #include <linux/vmalloc.h>
  #include <asm/io.h>
+ #include <media/v4l2-dev.h>
  
  #include "pwc.h"
  #include "pwc-ioctl.h"
***************
*** 128,134 ****
  	    .step    = 1,
  	    .default_value = 0,
  	},
! #if XAWTV_HAS_BEEN_FIXED
          {
  	    .id      = V4L2_CID_PRIVATE_SAVE_USER,
  	    .type    = V4L2_CTRL_TYPE_BUTTON,
--- 129,135 ----
  	    .step    = 1,
  	    .default_value = 0,
  	},
! #if 1 // XAWTV_HAS_BEEN_FIXED
          {
  	    .id      = V4L2_CID_PRIVATE_SAVE_USER,
  	    .type    = V4L2_CTRL_TYPE_BUTTON,
